invokeai/backend/model_manager/config.py

```
@@ -238,11 +257,11 @@ def classify(model_path: Path, hash_algo: HASHING_ALGORITHMS = "blake3_single",
 
         for config_cls in sorted_by_match_speed:
             try:
-                return config_cls.from_model_on_disk(mod, **overrides)
-            except InvalidModelConfigException:
-                logger.debug(f"ModelConfig '{config_cls.__name__}' failed to parse '{mod.path}', trying next config")
+                if config_cls.matches(mod):
```

It looks like if `config_cls.matches()` raises, we will bail out of the new API and fall back to the legacy probe (`ModelInstallService._probe()` handles the fallback when `ModelConfigBase.classify()` raises).

For example, in `LlavaOnevisionConfig.matches()`, I see a couple ways this could raise:
- `JSONDecodeError` when parsing the JSON
- `KeyError`, `IndexError`, etc when indexing into `architectures`

Should we restructure the exception handling here to avoid this situation?

---

@psychedelicious 
I agree. If `matches` throws, it indicates a runtime error (or a  bug), but that does not preclude another config type from matching successfully, thus we should continue searching.

On the other hand, if `from_model_on_disk` throws we should let the exception propagate: either the model is malformed, it's valid but we don't recognise it, we have matched the wrong config or we have a bug.

---

@psychedelicious
Thanks for the review! I've pushed commits to address your feedback.

---

Np.

Thinking about the UX - maybe an unhandled error while matching should be logged at warning level.

For example, consider a model that is not expected to match. While attempting the match, there is an unhandled exception. There's nothing wrong, and the user need need be alarmed.

I think the logging is useful for troubleshooting, though, so demoting it to warning could be a good middle ground.